lib/service/storage/elasticsearch.js

```diff
@@ -2388,12 +2398,46 @@ class ElasticSearch extends Service {
 
     return false;
   }
+
+  /**
+   * Checks if the dynamic properties are correct
+   */
+  _checkDynamicProperty (mappings) {
+    const dynamicProperties = findDynamic(mappings);
+    for (const [path, value] of Object.entries(dynamicProperties)) {
+      // Prevent common mistake
+      if (typeof value === 'boolean') {
+        _.set(mappings, path, value.toString());
+      }
+      else if (typeof value !== 'string') {
+        throw kerror.get(
+          'invalid_mapping',
+          path,
+          'Dynamic property value should be a string.');
```

Comment:
```suggestion
          'Dynamic property value must be a string.');
```

Reply:
well... not exactly since booleans are also accepter.

Perhaps this message would be better: `'The value of the "dynamic" property must be either a string or a boolean (accepted values: "true", "false", "strict")'`

Reply:
Actually even if we accept boolean as a common mistake, users should use the string version otherwise sometime they will send boolean for `true` and `false` and sometimes string for `strict`.  
So if they send a boolean it will be silently converted but any other value with result in an error

Reply:
Personally, I like unsurprising and exhaustive APIs. So if we accept boolean values, we should state so. If we don't, then we should reject them.

IMO, as a user, if I submit a boolean by mistake and Kuzzle doesn't complain, I'll go check the documentation. If I don't find anything about booleans being allowed, I'd be very confused... and think that the API is probably wonky.